### PR TITLE
QUAD-147: Added test cases for getAllProjectsDTOByAccessibility

### PR DESCRIPTION
**Pull request by sarvani11**
@jdamerow Added test cases. Please review and let us know your comments.

### CODE REVIEW

File: Quadriga/src/test/java/edu/asu/spring/quadriga/service/impl/workbench/RetrieveProjectManagerTest.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
+package edu.asu.spring.quadriga.service.impl.workbench;
```
Comment (jdamerow) [inline Quadriga/src/test/java/edu/asu/spring/quadriga/service/impl/workbench/RetrieveProjectManagerTest.java:1]: code style

File: Quadriga/src/test/java/edu/asu/spring/quadriga/service/impl/workbench/RetrieveProjectManagerTest.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+	private IProjectShallowMapper mockedProjectShallowMapper = Mockito.mock(IProjectShallowMapper.class);
+
+	@Before
+	public void setUp() throws QuadrigaStorageException {
+		MockitoAnnotations.initMocks(this);
+	}
+
+	@Test
+	public void checkOnlyPublicProjectsAreReturned() throws QuadrigaStorageException{
+		Mockito.when(mockedDBConnect.getAllProjectsDTOByAccessibility("PUBLIC")).thenReturn(createPublicProjectDTOList());
```
Comment (jdamerow) [inline Quadriga/src/test/java/edu/asu/spring/quadriga/service/impl/workbench/RetrieveProjectManagerTest.java:43]: you also want to make sure that this method is called. use mockito's verify method.

File: Quadriga/src/test/java/edu/asu/spring/quadriga/service/impl/workbench/RetrieveProjectManagerTest.java [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+		Mockito.when(mockedProjectShallowMapper.getProjectDetails(Matchers.any(ProjectDTO.class)))
+		.thenReturn(createPublicIProject());
+		List<IProject> projectList = retrieveProjectManagerUnderTest.getProjectListByAccessibility("PUBLIC");
+
+		assertNotNull(projectList);
+		assertEquals(1, projectList.size());
+		assertEquals("test Project1", projectList.get(0).getProjectName());
```
Comment (jdamerow) [inline Quadriga/src/test/java/edu/asu/spring/quadriga/service/impl/workbench/RetrieveProjectManagerTest.java:50]: you wnat to make sure that the projects you get are actually "public", the name is not that important.

File: Quadriga/src/test/java/edu/asu/spring/quadriga/service/impl/workbench/RetrieveProjectManagerTest.java [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+		Mockito.when(mockedDBConnect.getAllProjectsDTOByAccessibility("PUBLIC")).thenReturn(new ArrayList<ProjectDTO>());
+		List<IProject> projectList = retrieveProjectManagerUnderTest.getProjectListByAccessibility("PUBLIC");
+
+		assertNotNull(projectList);
+		assertEquals(0, projectList.size());
+	}
+	
+	@Test
+	public void checkProjectListByAccessibilityReturnsEmptyListIfNull() throws QuadrigaStorageException{
+		Mockito.when(mockedDBConnect.getAllProjectsDTOByAccessibility(null)).thenReturn(new ArrayList<ProjectDTO>());
```
Comment (jdamerow) [inline Quadriga/src/test/java/edu/asu/spring/quadriga/service/impl/workbench/RetrieveProjectManagerTest.java:64]: actually this doesn't make sense. why would you get an empty list if you supply bad parameters? I also don't think that this is what would actually happen in the code. I assume Hibernate would throw an exception. Just delete this test.

File: Quadriga/src/test/java/edu/asu/spring/quadriga/service/impl/workbench/RetrieveProjectManagerTest.java (hunk 1 above)
Comment (MeghanTalasila) [inline Quadriga/src/test/java/edu/asu/spring/quadriga/service/impl/workbench/RetrieveProjectManagerTest.java:1]: can you please let us know where exactly is the issue?

**Pull request closed without merging**